website/static/js/registrationUtils.js

```diff
@@ -15,51 +15,214 @@ var language = osfLanguage.registrations;
 
 var SaveManager = require('js/saveManager');
 var editorExtensions = require('js/registrationEditorExtensions');
-var registrationEmbargo = require('js/registrationEmbargo');
+var RegistrationModal = require('js/registrationModal');
+
+// This value should match website.settings.DRAFT_REGISTRATION_APPROVAL_PERIOD
+var DRAFT_REGISTRATION_MIN_EMBARGO_DAYS = 10;
+var DRAFT_REGISTRATION_MIN_EMBARGO_TIMESTAMP = new Date().getTime() + (
+        DRAFT_REGISTRATION_MIN_EMBARGO_DAYS * 24 * 60 * 60 * 1000
+);
+
+/**
+ * @class Comment
+ * Model for storing/editing/deleting comments on form fields
+ *
+ * @param {Object} data: optional data to instatiate model with
+ * @param {User} data.user
+ * @param {Date string} data.lastModified
+ * @param {Date string} data.created
+ * @param {Boolean} data.isDeleted
+ * @param {String} data.value
+ *
+ * @type User
+ * @property {String} id
+ * @property {String} fullname
+ **/
+function Comment(data) {
+    var self = this;
+
+    self.saved = ko.observable(data ? true : false);
+
+    data = data || {};
+    self.user = data.user || $osf.currentUser();
+    self.value = ko.observable(data.value || '');
+    self.value.subscribe(function() {
+        self.lastModified = new Date();
+    });
+
+    self.created = data.created ? new Date(data.created) : new Date();
+    self.lastModified = data.lastModified ? new Date(data.lastModified) : new Date();
+
+    self.isDeleted = ko.observable(data.isDeleted || false);
+    self.isDeleted.subscribe(function(isDeleted) {
+        if (isDeleted) {
+            self.value('');
+        }
+    });
+
+    self.seenBy = ko.observableArray([self.user.id]);
+
+    /**
+     * Returns true if the current user is the comment owner
+     **/
+    self.isOwner = ko.pureComputed(function() {
+        return self.user.id === $osf.currentUser().id;
+    });
+
+    /**
+     * Returns the author as the actual user, not 'You'
+     **/
+    self.author = ko.pureComputed(function() {
+        return self.user.fullname;
```

Comment:
This (and probably others in this file)  does not need to be a `computed`.

Use a computed observable when the value depends on something that changes (another observable). This just depends on a static value, so there's no reason for knockout to keep watching it.

http://knockoutjs.com/documentation/computedObservables.html
> That’s where computed observables come in - these are functions that are dependent on one or more other observables, and will automatically update whenever any of these dependencies change.